src/category_theory/sites/sieves.lean

```diff
@@ -355,8 +355,24 @@ lemma nat_trans_of_le_comm {S T : sieve X} (h : S ≤ T) :
 rfl
 
 /-- The presheaf induced by a sieve is a subobject of the yoneda embedding. -/
-instance functor_inclusion_is_mono : mono (functor_inclusion S) :=
+instance functor_inclusion_is_mono : mono S.functor_inclusion :=
 ⟨λ Z f g h, by { ext Y y, apply congr_fun (nat_trans.congr_app h Y) y }⟩
 
+/-- A natural transformation to a representable functor induces a sieve. -/
+-- TODO: show that when `f` is mono, this is inverse to `functor_inclusion`, in the appropriate
```

Comment:
How hard would it be to include this TODO in this PR. Is it hard?

Reply:
One direction is fine, I've just included it; the other direction is awkwardly subtle (in particular, going from `R -> yoneda.obj X` to a sieve then back to the original arrow. One reason is that it's not a genuine inverse, but an inverse up to isomorphism (in the comma category, or in the category of subobjects), another is that to show this inverse we need that the components of a monomorphism in a functor category over Type are all monos (which is why I made those PRs about mono iff pullback, but there are one or two more small bits needed to get there). On top of this, I don't think that direction is particularly helpful (though I might change my mind as I keep doing the sheaf stuff), but I'm also working out a PR to add the category (poset) of subobjects, and I think the *right* way of stating this bijection is to say a sieve on X is the same thing as a subobject of `yoneda.obj X`; since the poset of subobjects is a skeletal category.